dfget/core/helper/helper_test.go

```diff
@@ -74,3 +79,114 @@ func (s *HelperTestSuite) TestDownloadPattern(c *check.C) {
 			check.Commentf("f:%v pattern:%s", name(v.f), v.pattern))
 	}
 }
+
+func (s *HelperTestSuite) readFromFileServer(port int, path string, off int64, size int64) ([]byte, error) {
+	url := fmt.Sprintf("http://127.0.0.1:%d/%s", port, path)
+	header := map[string]string{}
+
+	if size > 0 {
+		header["Range"] = fmt.Sprintf("bytes=%d-%d", off, off+size-1)
+	}
+
+	resp, err := httputils.HTTPGet(url, header)
+	if err != nil {
+		return nil, err
+	}
+
+	if resp.StatusCode >= 400 {
+		return nil, fmt.Errorf("resp code %d", resp.StatusCode)
+	}
+
+	defer resp.Body.Close()
```

Comment:
I am afraid that you should move this `defer` in front of the `400` judgement

Reply:
done.